src/main/kotlin/org/rust/lang/core/completion/RsBoolCompletionProvider.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.completion
+
+import com.intellij.codeInsight.completion.CompletionParameters
+import com.intellij.codeInsight.completion.CompletionResultSet
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.patterns.ElementPattern
+import com.intellij.psi.PsiElement
+import com.intellij.util.ProcessingContext
+import org.rust.lang.core.RsPsiPattern
+import org.rust.lang.core.psi.RsPathExpr
+import org.rust.lang.core.psi.ext.ancestorOrSelf
+import org.rust.lang.core.types.expectedType
+import org.rust.lang.core.types.ty.TyBool
+
+object RsBoolCompletionProvider : RsCompletionProvider() {
+
+    override val elementPattern: ElementPattern<PsiElement> get() = RsPsiPattern.simplePathPattern
+
+    override fun addCompletions(parameters: CompletionParameters, context: ProcessingContext, result: CompletionResultSet) {
+        if (parameters.position.ancestorOrSelf<RsPathExpr>()?.expectedType == TyBool) {
```

Comment:
```suggestion
        if (parameters.position.safeGetOriginalOrSelf().ancestorOrSelf<RsPathExpr>()?.expectedType == TyBool) {
```

`parameters.position` returns psi element in copy of original PSI file. `safeGetOriginalOrSelf()` returns element from original file if possible. It allows us to reuse existing type inference/name resolution caches if possible

Reply:
Done!